api_client/security.py

```diff
@@ -65,22 +65,30 @@ def close(self):
         self.revoke_auth_token()
 
     def request_new_auth_token(self):
-        url_path = '/sso-api/v1/token'
-        url = urllib.parse.urljoin(self.sso_svcs_base_url, url_path)
+        if hasattr(self, 'user_id') and hasattr(self, 'user_password'):
+            url_path = '/sso-api/v1/token'
+            url = urllib.parse.urljoin(self.sso_svcs_base_url, url_path)
+            request_new_auth_token_data = {
+                'username': self.user_id,
+                'password': self.user_password,
+                'grant_type': 'password',
+                'scope': 'openid'
+            }
+            response = requests.post(
+                url,
+                data=request_new_auth_token_data,
+                auth=(self.user_id, self.user_password),
+                proxies=self.proxies
+            )
+        else: 
```

Comment:
The code below should be at the method `def renew_auth_token(self): `https://github.com/moodysanalytics/apic/blob/cdf20d503f6797efcd0f0882ada561795f23fc6a/api_client/security.py#L115
the current method `request_new_auth_token` should be left intact

Reply:
Done